src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -700,6 +704,97 @@ private string DebuggerDisplayMethodDescription
             }
         }
 
+        #region EventSource Tracing
+
+        void EtwTaskStarted(Task previousTask, ref Guid savedActivityID)
+        {
+            TplEtwProvider etwLog = TplEtwProvider.Log;
+            if (etwLog.TasksSetActivityIds)
+                EventSource.SetCurrentThreadActivityId(TplEtwProvider.CreateGuidForTaskID(this.Id), out savedActivityID);
+            // previousTask holds the actual "current task" we want to report in the event
+            if (previousTask != null)
+                etwLog.TaskStarted(previousTask.m_taskScheduler.Id, previousTask.Id, this.Id);
+            else
+                etwLog.TaskStarted(TaskScheduler.Current.Id, 0, this.Id);
+        }
+
+        void EtwTaskCompleted(Task previousTask, ref Guid savedActivityID)
```

Comment:
Why did all of these need to be refactored into helpers?

Reply:
It's not needed for this PR it's just way how the PR evolved and I found it useful to have it separated similar way as other tracing mechanisms in Task. Some of them could possibly be reused by TaskContinuation but I didn't look into that yet.